### PR TITLE
ci: use build-in yarn cache

### LINKED ISSUE #10706: Use built-in caching that comes with `actions/setup-node@v2.2.0`
**What would you like Renovate to be able to do?**

I think/hope we can simplify our cache setup by using the new built-in cache that comes with `actions/setup-node@v2.2.0`.

**Did you already have any implementation ideas?**

The version of `actions/setup` we need is already live on our `main` branch.

But work is needed to change the old cache method to the new method.

Release notes and manual: https://github.com/actions/setup-node/releases/tag/v2.2.0

Normally I'd offer to do this myself, but the current CI setup is too complicated for me to understand. So I'm going to sit tight and let somebody who really understands GitHub Actions make this change. 😄 

I don't know if we can satisfy _all_ of our Yarn caching needs though:

> Yarn caching handles both yarn versions: 1 or 2.
> 
>     At the moment, only `lock` files in the project root are supported.

I don't know if this would be enough or not, just mentioning it here.

### PR DESCRIPTION
**Pull request by viceice**
## Changes:

Use build-in caching of `actions/setup-node`.

## Context:

closes #10706

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -11,13 +11,11 @@ on:
       dryRun:
         description: 'Dry-Run'
         default: 'true'
+        required: false
```
Comment (HonkingGoose) [inline .github/workflows/build.yml:14]: There's a behavior/logic change here. What's the idea here?

File: .github/workflows/release-npm.yml [hunk 2]
```diff
@@ -15,11 +15,9 @@ on:
       tag:
         description: 'Npm dist-tag'
         default: 'latest'
+        required: false
```
Comment (HonkingGoose) [inline .github/workflows/release-npm.yml:18]: Logic change here as well.

File: .github/workflows/release-npm.yml [hunk 3]
```diff
@@ -55,12 +39,20 @@ jobs:
...
+          cache: yarn
+
+      - name: Init platform
+        run: |
+          git config --global core.autocrlf false
+          git config --global core.symlinks true
+          git config --global user.email 'renovate@whitesourcesoftware.com'
+          git config --global user.name  'Renovate Bot'
+          yarn config set version-git-tag false
+          npm config set scripts-prepend-node-path true
```
Comment (HonkingGoose) [inline .github/workflows/release-npm.yml:55]: This is also new?

**Review by HonkingGoose — COMMENTED** [review]
Cache stuff looks okay, and it seems to work fine on the action log (it says: "Cache not found", but that's because there's no cache to find right now... 😉).

I do notice some logic/behavior changes, that I have some questions about.

File: .github/workflows/build.yml (hunk 1 above)
Comment (viceice) [reply .github/workflows/build.yml:14]: github action validation in vscode told me that this field is required, so i've added it with false value to be optional.

File: .github/workflows/release-npm.yml (hunk 2 above)
Comment (viceice) [reply .github/workflows/release-npm.yml:18]: no logic change, as default value for `required` is false if not set, see github docs

File: .github/workflows/release-npm.yml (hunk 3 above)
Comment (viceice) [reply .github/workflows/release-npm.yml:55]: not new, just some reordering, we need to checkout first, so `actions/setup-node` can see the lock file.

**Review by HonkingGoose — APPROVED** [review]

**Review by rarkins — APPROVED** [review]

**Comment by HonkingGoose** [comment]
@viceice The `release` action is complaining now this is merged into the `main` branch:

```
Post job cleanup.
/usr/local/bin/yarn --version
1.22.10
/usr/local/bin/yarn cache dir
error An unexpected error occurred: "Failed to replace env in config: ${NPM_TOKEN}".
info If you think this is a bug, please open a bug report with the information provided in "/home/runner/work/renovate/renovate/yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/cache for documentation about this command.
Error: The process '/usr/local/bin/yarn' failed with exit code 1
```

Link to action error message: https://github.com/renovatebot/renovate/runs/2990123628#step:13:5

**Comment by viceice** [comment]
@HonkingGoose see #10715

**Pull request merged**